src/wp-includes/rest-api/endpoints/class-wp-rest-posts-controller.php

```diff
@@ -1012,6 +1049,8 @@ protected function prepare_items_query( $prepared_args = array(), $request = nul
 			 *
 			 * @param string $value The query_var value.
 			 */
+			// TODO: Do not merge without auditing filters
+			// BREAKING CHANGE! Other filters too? orderby was a string but is now a PHP array (either a REST array or a REST object).
```

Comment:
Highlighting that we still need to 🤔 audit existing filters

Reply:
For the `rest_query_var-orderby` filter I do think it's clearest to simply pass through value, which might be a string or array. 

If we'd really like to preserve backwards compatibility, we could do something like pass through the first argument as a string (maybe the first value or some comma or space delimited string), and have the second argument be the original value. I feel like it'd avoid potential errors on upgrade, but would be confusing for future usage.

Reply:
Taking another look, it looks like the default implementation returns $value so passing through a string as a first arg won't work.